app/src/main/java/com/alphawallet/app/ui/widget/holder/WalletSummaryHolder.java

```diff
@@ -220,6 +281,29 @@ private Wallet fetchWallet(Wallet w)
             w.ENSAvatar = realmWallet.getENSAvatar();
         }
 
```

Comment:
Use the 'Token[]' array from WalletsActivity to populate this. No need for fetching anything from tokensService or tokenRepository. You may have to sort the Token[] array.